charts/fleet/templates/deployment.yaml

```
@@ -18,55 +24,59 @@ spec:
         - name: FLEET_PROPAGATE_DEBUG_SETTINGS_TO_AGENTS
-          value: {{ quote .Values.propagateDebugSettingsToAgents }}
-        {{- if .Values.clusterEnqueueDelay }}
+          value: {{ quote $.Values.propagateDebugSettingsToAgents }}
+        {{- if $.Values.clusterEnqueueDelay }}
         - name: FLEET_CLUSTER_ENQUEUE_DELAY
-          value: {{ .Values.clusterEnqueueDelay }}
+          value: {{ $.Values.clusterEnqueueDelay }}
         {{- end }}
-        {{- if .Values.proxy }}
+        {{- if $.Values.proxy }}
         - name: HTTP_PROXY
-          value: {{ .Values.proxy }}
+          value: {{ $.Values.proxy }}
         - name: HTTPS_PROXY
-          value: {{ .Values.proxy }}
+          value: {{ $.Values.proxy }}
         - name: NO_PROXY
-          value: {{ .Values.noProxy }}
+          value: {{ $.Values.noProxy }}
         {{- end }}
-        {{- if .Values.cpuPprof }}
+        {{- if $.Values.cpuPprof }}
         - name: FLEET_CPU_PPROF_DIR
           value: /tmp/pprof/
         {{- end }}
-        {{- if .Values.cpuPprof }}
+        {{- if $.Values.cpuPprof }}
         - name: FLEET_CPU_PPROF_PERIOD
-          value: {{ quote .Values.cpuPprof.period }}
+          value: {{ quote $.Values.cpuPprof.period }}
         {{- end }}
-        {{- if .Values.leaderElection.leaseDuration }}
+        {{- if $.Values.leaderElection.leaseDuration }}
         - name: CATTLE_ELECTION_LEASE_DURATION
-          value: {{.Values.leaderElection.leaseDuration}}
+          value: {{$.Values.leaderElection.leaseDuration}}
         {{- end }}
-        {{- if .Values.leaderElection.retryPeriod }}
+        {{- if $.Values.leaderElection.retryPeriod }}
         - name: CATTLE_ELECTION_RETRY_PERIOD
-          value: {{.Values.leaderElection.retryPeriod}}
+          value: {{$.Values.leaderElection.retryPeriod}}
         {{- end }}
-        {{- if .Values.leaderElection.renewDeadline }}
+        {{- if $.Values.leaderElection.renewDeadline }}
         - name: CATTLE_ELECTION_RENEW_DEADLINE
-          value: {{.Values.leaderElection.renewDeadline}}
+          value: {{$.Values.leaderElection.renewDeadline}}
         {{- end }}
-        {{- if .Values.debug }}
+        {{- if $.Values.debug }}
         - name: CATTLE_DEV_MODE
           value: "true"
         {{- end }}
-        image: '{{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag }}'
+        image: '{{ template "system_default_registry" $ }}{{ $.Values.image.repository }}:{{ $.Values.image.tag }}'
         name: fleet-controller
-        imagePullPolicy: "{{ .Values.image.imagePullPolicy }}"
+        imagePullPolicy: "{{ $.Values.image.imagePullPolicy }}"
         command:
         - fleetcontroller
-        {{- if not .Values.gitops.enabled }}
+        {{- if not $.Values.gitops.enabled }}
         - --disable-gitops
         {{- end }}
-        {{- if .Values.debug }}
+        {{- if . }}
+        - --shard-id
+        - {{ . }}
```

```suggestion
        - {{ quote . }}
```

Or we document id has to start with a letter.

---

Good point, fixed, thanks!